File: data/code_pairs/cpp/278.cpp
#include <vector>
#include <cstdint>

class Mod_MBVariables {
public:
    int32_t incp;

    std::vector<double> Nslv, dNslvdxi, dNslv2dxi2;
    std::vector<double> Mslv, dMslvdeta, dMslv2deta2;
    std::vector<double> R1;
    std::vector<std::vector<double>> dR1, ddR1;
    std::vector<double> Rslv, gap;
    std::vector<std::vector<double>> dRslv, ddRslv;
    std::vector<double> Rmst;
    std::vector<std::vector<double>> dRmst, ddRmst;
    std::vector<int32_t> PTS_Active;

    std::vector<std::vector<double>> N, Na, Nb, Ta, tb;
    std::vector<std::vector<double>> Nhat, That, D, Nbar;
    std::vector<std::vector<double>> Kgeo, KG1, KG2, KG3, KG4;

    std::vector<std::vector<double>> KLM, FLM, GLM;
    std::vector<int32_t> disptempLM;
    std::vector<std::vector<double>> KeqLM, FeqLM, DispeqLM;

    std::vector<int32_t> PTS_conn;
};

File: data/code_pairs/fortran/278.f90
module Mod_MBVariables
    implicit none
    integer(4)::incp
    
    real(8), dimension(:), allocatable :: Nslv, dNslvdxi, dNslv2dxi2
    real(8), dimension(:), allocatable :: Mslv, dMslvdeta, dMslv2deta2
    real(8), dimension(:), allocatable :: R1
    real(8), dimension(:,:), allocatable :: dR1, ddR1
    real(8), dimension(:), allocatable :: Rslv, gap
    real(8), dimension(:,:), allocatable :: dRslv, ddRslv
    real(8), dimension(:), allocatable :: Rmst
    real(8), dimension(:,:), allocatable :: dRmst, ddRmst
    integer(4), dimension(:), allocatable :: PTS_Active
    
    real(8), dimension(:,:), allocatable :: N, Na, Nb, Ta, tb
    real(8), dimension(:,:), allocatable :: Nhat, That, D, Nbar
    real(8), dimension(:,:), allocatable :: Kgeo, KG1, KG2, KG3, KG4
    
    real(8), dimension(:,:), allocatable :: KLM, FLM, GLM
    integer(4), dimension(:), allocatable :: disptempLM
    real(8), dimension(:,:), allocatable :: KeqLM, FeqLM, DispeqLM
    
    integer(4), dimension(:), allocatable :: PTS_conn
end module Mod_MBVariables